src/Agent.Worker/TestResults/ContainerStructureTestResultReader.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    /// <summary>
+    /// Reads result output from google container strucutre test.
+    /// https://github.com/GoogleContainerTools/container-structure-test
+    /// Example JSON: https://gist.github.com/navin22/30edd4041f5eb14b0d860ee07fdc2184
+    /// </summary>
+    public class ContainerStructureTestResultReader : AgentService, IResultReader
+    {
+        public bool AddResultsFileToRunLevelAttachments {get;set;}
+
+        public string Name => "ContainerStructure";
+
+        public Type ExtensionType => typeof(IResultReader);
+
+        public ContainerStructureTestResultReader()
+        {
+            AddResultsFileToRunLevelAttachments = true;
+        }
+
+        public TestRunData ReadResults(IExecutionContext executionContext, string filePath, TestRunContext runContext)
+        {
+            try
+            {
+                string jsonTestSummary = File.ReadAllText(filePath);
+                if (string.IsNullOrWhiteSpace(jsonTestSummary)) 
+                {
+                    return null;
+                }
+
+                JsonTestSummary testSummary = StringUtil.ConvertFromJson<JsonTestSummary>(jsonTestSummary);
```

Comment:
we should be carefull here to verify we are trying to jsonify valid file, please add catch and pre validation check if possible.

Reply:
This is under a try a catch. will not be affecting the task. 
Since we don't have a proper schema for the json, I will go ahead with this.